In [1]:
import pandas as pd

In [27]:
songs = pd.read_csv('datasets/X_test_predictions.csv')

In [28]:
songs.head()

clean_lyrics  american_songbook  \
0  every time i look down on this timeless town w...                  1   
1  romance is a game for fools i used to say a ga...                  1   
2  liza liza skies are gray but if you smile on m...                  1   
3  i'm so unhappy what'll i do i long for somebod...                  1   
4  everything is set skies are blue can't believe...                  1   

       song_writer  predictions  disney_renaissance  \
0      Cole Porter            1                   0   
1    Johnny Mercer            1                   0   
2  George Gershwin            1                   0   
3    Irving Berlin            1                   0   
4    Johnny Mercer            1                   0   

                                              lyrics  \
0  \n                    Every time I look down o...   
1  \n                    "Romance is a game for f...   
2  \n                    Liza, Liza, skies are gr...   
3  \n                    I'm so unhappy\nWhat'll ...   
4  \n                    Everything is set, skies...   

                                    song_name  
0  I Love Paris, Song (from "Can Can") Lyrics  
1                        Fools Rush In Lyrics  
2   Liza (All The Clouds'll Roll Away) Lyrics  
3                        All By Myself Lyrics  
4                       Sweet Lorraine Lyrics

In [29]:
songs.drop(columns=['lyrics'], inplace=True)

In [61]:
songs.shape

(204, 6)

In [30]:
songs['disney_renaissance'].value_counts()

0    168
1     36
Name: disney_renaissance, dtype: int64

In [31]:
songs['american_songbook'].value_counts(normalize=True)

1    0.544118
0    0.455882
Name: american_songbook, dtype: float64

In [32]:
songs['american_songbook'].value_counts()

1    111
0     93
Name: american_songbook, dtype: int64

In [47]:
36 / 93

0.3870967741935484

Disney Renaissance makes up 38.7% of the Disney songs, but 50% of the false predictions

In [34]:
songs['predictions'].value_counts(normalize=True)

1    0.612745
0    0.387255
Name: predictions, dtype: float64

In [37]:
false_predictions = songs.loc[~(songs['predictions'] == songs['american_songbook']), :]

In [39]:
false_predictions['disney_renaissance'].value_counts(normalize=True)

0    0.647059
1    0.352941
Name: disney_renaissance, dtype: float64

In [41]:
false_predictions['american_songbook'].value_counts(normalize=True)

0    0.705882
1    0.294118
Name: american_songbook, dtype: float64

In [64]:
false_predictions['american_songbook'].value_counts()

0    24
1    10
Name: american_songbook, dtype: int64

In [48]:
false_predictions['predictions'].value_counts(normalize=True)

1    0.705882
0    0.294118
Name: predictions, dtype: float64

In [43]:
false_predictions.shape

(34, 6)

In [45]:
songs.shape

(204, 6)

In [55]:
disney_false = false_predictions.loc[false_predictions['american_songbook'] == 0, :]

In [59]:
disney_false['disney_renaissance'].value_counts(normalize=True)

1    0.5
0    0.5
Name: disney_renaissance, dtype: float64

In [60]:
disney_false

clean_lyrics  american_songbook  \
13   in the quiet time of evening when the stars as...                  0   
14   verse there is something i got to tell you i g...                  0   
18   shang let's get down to business to defeat the...                  0   
22   oh the power to be strong and the wisdom to be...                  0   
29   following the leader the leader the leader we'...                  0   
31   once upon a time in a faraway land a young pri...                  0   
37   reindeers are better than people sven don't yo...                  0   
47   i've got my mother's love i shouldn't ask for ...                  0   
53   'twas brillig and the slithy toves did gyre an...                  0   
57   you think i'm just an ignorant savage and you'...                  0   
81   oh this is the night it's a beautiful night an...                  0   
83   verse your gaze got me hypnotized hypnotized y...                  0   
97   casey junior's comin' down the track comin' do...                  0   
102  lady tremaine the pear-shaped tone drizella oh...                  0   
118  tale as old as time true as it can be barely e...                  0   
119  baby mine don't you cry baby mine dry your eye...                  0   
131  ariel well dear friend it's time to put this l...                  0   
146  love me hug me if you wanna touch me kiss me r...                  0   
148  paris the city of lovers is glowing this eveni...                  0   
166  hakuna matata what a wonderful phrase hakuna m...                  0   
173  with one wave of my hand you know who's in com...                  0   
177  cruella de vil cruella de vil if she doesn't s...                  0   
190  maybe you've heard of a terrible place where t...                  0   
191  aurora borealis the icy sky at night our paddl...                  0   

                     song_writer  predictions  disney_renaissance  \
13      The Emperor's New Groove            1                   0   
14                       Aladdin            1                   1   
18                         Mulan            1                   1   
22                        Tarzan            1                   1   
29                     Peter Pan            1                   0   
31          Beauty and the Beast            1                   1   
37                        Frozen            1                   0   
47                       Tangled            1                   0   
53           Alice In Wonderland            1                   0   
57                    Pocahontas            1                   1   
81            Lady and the Tramp            1                   0   
83                    Pocahontas            1                   1   
97                         Dumbo            1                   0   
102                   Cinderella            1                   0   
118         Beauty and the Beast            1                   1   
119                        Dumbo            1                   0   
131               Little Mermaid            1                   1   
146               Little Mermaid            1                   1   
148  The Hunchback of Notre Dame            1                   0   
166                    Lion King            1                   1   
173               Little Mermaid            1                   1   
177               101 Dalmations            1                   0   
190  The Hunchback of Notre Dame            1                   0   
191                   Pocahontas            1                   1   

                                    song_name  
13                     My Funny Friend and Me  
14               My Love Is Not A Game Lyrics  
18               Make A Man Out Of You Lyrics  
22                          Son Of Man Lyrics  
29                       Following the Leader  
31          Prologue (The Enchantress) Lyrics  
37         Reindeer(s) Are Better Than People  
47    

In [46]:
false_predictions

clean_lyrics  american_songbook  \
13   in the quiet time of evening when the stars as...                  0   
14   verse there is something i got to tell you i g...                  0   
18   shang let's get down to business to defeat the...                  0   
22   oh the power to be strong and the wisdom to be...                  0   
24   people are queer they're always crowing scramb...                  1   
29   following the leader the leader the leader we'...                  0   
31   once upon a time in a faraway land a young pri...                  0   
37   reindeers are better than people sven don't yo...                  0   
47   i've got my mother's love i shouldn't ask for ...                  0   
53   'twas brillig and the slithy toves did gyre an...                  0   
57   you think i'm just an ignorant savage and you'...                  0   
63   you're lonely and i'm lonely so why can't we b...                  1   
81   oh this is the night it's a beautiful night an...                  0   
83   verse your gaze got me hypnotized hypnotized y...                  0   
88   hold tight hold tight hold tight hold tight fo...                  1   
97   casey junior's comin' down the track comin' do...                  0   
102  lady tremaine the pear-shaped tone drizella oh...                  0   
107  as you listen to the band don't you get a bubb...                  1   
113  give me your tired your poor your huddled mass...                  1   
118  tale as old as time true as it can be barely e...                  0   
119  baby mine don't you cry baby mine dry your eye...                  0   
131  ariel well dear friend it's time to put this l...                  0   
146  love me hug me if you wanna touch me kiss me r...                  0   
148  paris the city of lovers is glowing this eveni...                  0   
151  do you hear that whistle down the line i figur...                  1   
161  we might have been meant for each other to be ...                  1   
162  y shining hour this moment this minute and eac...                  1   
166  hakuna matata what a wonderful phrase hakuna m...                  0   
173  with one wave of my hand you know who's in com...                  0   
176  charted by frank at in charted by eddy howard ...                  1   
177  cruella de vil cruella de vil if she doesn't s...                  0   
190  maybe you've heard of a terrible place where t...                  0   
191  aurora borealis the icy sky at night our paddl...                  0   
202  somewhere someday we'll be close together wait...                  1   

                     song_writer  predictions  disney_renaissance  \
13      The Emperor's New Groove            1                   0   
14                       Aladdin            1                   1   
18                         Mulan            1                   1   
22                        Tarzan            1                   1   
24               George Gershwin            0                   0   
29                     Peter Pan            1                   0   
31          Beauty and the Beast            1                   1   
37                        Frozen            1                   0   
47                       Tangled            1                   0   
53           Alice In Wonderland            1                   0   
57                    Pocahontas            1                   1   
63                 Irving Berlin            0                   0   
81            Lady and the Tramp            1                   0   
83                    Pocahontas            1                   1   
88                   Fats Waller            0                   0   
97                         Dumbo            1                   0   
102                   Cinderella            1                   0   
107                Irving Berlin            0                   0   
113                Irving Berlin   

In [49]:
# Mulan Make a Man out of You/
false_predictions.iloc[2]['clean_lyrics']

"shang let's get down to business to defeat the huns did they send me daughters when i asked for sons you're the saddest bunch i've ever met but you can bet before were through mister i'll make a man out of you tranquil as a forest but on fire within once you find your center you are sure to win you're a spineless pale pathetic lot and you haven't got a clue somehow i'll make a man out of you chan-po i'm never gonna catch my breath yow say goodbye to those who knew me ling why was i a fool in school for cutting gym mushu this guy's got them scared to death mulan ping hope he doesn't see right through me chan-po now i really wish that i knew how to swim be a man we must be swift as the coursing river be a man with all the force of a great typhoon be a man with all the strength of a raging fire mysterious as the dark side of the moon time is racing toward us 'til the huns arrive heed my every order and you might survive you're unsuited for the rage of war so pack up go home you're throug

In [52]:
# The Emperor's New Groove My Funny Friend and Me
false_predictions.iloc[0]['clean_lyrics']

"in the quiet time of evening when the stars assume their patterns and the day has made his journey and we wonder just what happened to the life we knew before the world changed when not a thing i held was true but you were kind to me and you reminded me that the world is not my playground there are other things that matter and what is simple needs protecting my illusions all would shatter but you stayed in my corner when the only world i knew was upside down and now the world and we will know you carried me you see the patterns in the big sky those constellations look like you and i just like the patterns in the big sky we could be lost we could refuse to try but we made it through in the dark night would those lucky guys turn out to be but that unusual blend of my funny friend and me i'm not as clever as i thought i was i'm not the boy i used to be because you showed me something different you showed me something pure i always seemed so certain but i was really never sure but you sta

In [67]:
# George Gershwin Life Is Just A Bowl Of Cherries Lyrics
false_predictions.iloc[4]['clean_lyrics']

"people are queer they're always crowing scrambling and rushing about why don't they stop someday address themselves this way why are we here where are we going it's time that we found out we're not here to stay we're on a short holiday life is just a bowl of cherries don't take it serious life's so mysterious you work you save you worry so but you can't take your dough when you go go go so keep repeating it's the berries the strongest oak must fall the sweet things in life to you were just loaned so how can you lose what you've never owned life is just a bowl of cherries so live and laugh at it all life is just a bowl of cherries don't take it serious it's too mysterious at eight each morning i have got a date to take my plunge 'round the empire state you'll admit it's not the berries in a building that's so tall there's a guy in the show the girls love to kiss get thousands a week just for crooning like this life is just a bowl of aw nuts so live and laugh at it all"

In [62]:
coefs = pd.read_csv('datasets/coefs_with_and_without_stopwords.csv')

In [63]:
coefs = coefs.rename(columns={'Unnamed: 0': 'words'})

In [64]:
coefs.head()

words     coefs  coefs_with_stopword
0        song -1.657070            -1.962134
1        this -1.557221                  NaN
2         see -1.423257                  NaN
3        each -1.274532                  NaN
4  everything -1.151166                  NaN

In [65]:
coefs['coef_differences'] = coefs['coefs'] - coefs['coefs_with_stopword']

In [66]:
coefs.head(20)

words     coefs  coefs_with_stopword  coef_differences
0         song -1.657070            -1.962134          0.305065
1         this -1.557221                  NaN               NaN
2          see -1.423257                  NaN               NaN
3         each -1.274532                  NaN               NaN
4   everything -1.151166                  NaN               NaN
5           is -1.077905                  NaN               NaN
6         will -1.075932                  NaN               NaN
7         life -1.069416            -1.049800         -0.019616
8      someone -1.051065                  NaN               NaN
9         king -1.046943            -1.012902         -0.034041
10     friends -1.042089            -0.745971         -0.296117
11          we -1.040986                  NaN               NaN
12        here -1.032910                  NaN               NaN
13         now -1.032278                  NaN               NaN
14         way -1.026359            -1.140443          0.114084
15         her -1.025607                  NaN               NaN
16      inside -1.011767            -1.137415          0.125648
17       proud -1.005710            -0.924672         -0.081037
18      gaston -0.975910            -1.007493          0.031583
19       belle -0.960768            -1.128200          0.167432

In [71]:
coefs.sort_values('coefs').head(50)

words     coefs  coefs_with_stopword  coef_differences
0         song -1.657070            -1.962134          0.305065
1         this -1.557221                  NaN               NaN
2          see -1.423257                  NaN               NaN
3         each -1.274532                  NaN               NaN
4   everything -1.151166                  NaN               NaN
5           is -1.077905                  NaN               NaN
6         will -1.075932                  NaN               NaN
7         life -1.069416            -1.049800         -0.019616
8      someone -1.051065                  NaN               NaN
9         king -1.046943            -1.012902         -0.034041
10     friends -1.042089            -0.745971         -0.296117
11          we -1.040986                  NaN               NaN
12        here -1.032910                  NaN               NaN
13         now -1.032278                  NaN               NaN
14         way -1.026359            -1.140443          0.114084
15         her -1.025607                  NaN               NaN
16      inside -1.011767            -1.137415          0.125648
17       proud -1.005710            -0.924672         -0.081037
18      gaston -0.975910            -1.007493          0.031583
19       belle -0.960768            -1.128200          0.167432
20        need -0.960626            -0.798915         -0.161711
21       there -0.958167                  NaN               NaN
22        come -0.951190            -0.812896         -0.138294
23        wish -0.950706            -1.084684          0.133977
24      second -0.944046            -0.915014         -0.029032
25          ha -0.940213            -0.855990         -0.084223
26    tuppence -0.936311            -0.915926         -0.020385
27          no -0.932008                  NaN               NaN
28     finally -0.924334            -0.983777          0.059443
29         ali -0.906400            -0.838822         -0.067578
30          la -0.900913            -1.024638          0.123725
31    remember -0.900538            -0.504907         -0.395631
32     aladdin -0.895719            -0.921055          0.025336
33       world -0.895658            -1.078975          0.183317
34         max -0.877618            -0.932609          0.054991
35     goodbye -0.863936            -1.117720          0.253784
36        chim -0.853659            -0.887293          0.033633
37      enough -0.848357                  NaN               NaN
38        even -0.841488                  NaN               NaN
39        path -0.838538            -1.151061          0.312523
40       makes -0.831353            -0.836872          0.005519
41        what -0.830016                  NaN               NaN
42      friend -0.828609            -0.791503         -0.037106
43      though -0.827233                  NaN               NaN
44      wonder -0.825329            -0.811415         -0.013914
45        very -0.824094                  NaN               NaN
46   believing -0.817739            -0.987285          0.169545
47     welcome -0.811055            -0.761237         -0.049818
48         bop -0.808798            -0.805022         -0.003776
49       guard -0.808502            -0.796166         -0.012336

In [72]:
coefs.sort_values('coefs', ascending=False).head(50)

words     coefs  coefs_with_stopword  coef_differences
2999        baby  1.612322             1.400892          0.211430
2998         was  1.450071                  NaN               NaN
2997         man  1.379970             1.048606          0.331364
2996        love  1.316975             1.303905          0.013070
2995        then  1.315654                  NaN               NaN
2994         old  1.307268             1.271998          0.035270
2993       white  1.265955             1.326396         -0.060442
2992        that  1.229510                  NaN               NaN
2991       shall  1.229362             0.794054          0.435308
2990         day  1.223301             0.950226          0.273076
2989        when  1.218216                  NaN               NaN
2988          by  1.189394                  NaN               NaN
2987         say  1.172162             1.550532         -0.378370
2986       night  1.150758             1.521041         -0.370283
2985          my  1.147584                  NaN               NaN
2984         may  1.146353                  NaN               NaN
2983         she  1.132515                  NaN               NaN
2982         but  1.128845                  NaN               NaN
2981        dear  1.120327             1.190570         -0.070243
2980       while  1.036993                  NaN               NaN
2979        from  1.019934                  NaN               NaN
2978       dance  0.999631             0.831959          0.167672
2977      afraid  0.999021             1.195563         -0.196543
2976        moon  0.998838             1.068738         -0.069900
2975       money  0.994797             0.962357          0.032440
2974         joe  0.965470             1.018108         -0.052638
2973          as  0.963081                  NaN               NaN
2972        doin  0.936078             0.787391          0.148687
2971         met  0.935345             1.129629         -0.194284
2970       woman  0.933002             0.951492         -0.018490
2969          to  0.932647                  NaN               NaN
2968        easy  0.921215             1.409904         -0.488688
2967       happy  0.908975             0.932676         -0.023701
2966        tell  0.900970             1.087209         -0.186239
2965         any  0.887392                  NaN               NaN
2964        high  0.884754             0.984252         -0.099498
2963    somebody  0.863425             0.812305          0.051121
2962        band  0.854098             0.920037         -0.065939
2961       smile  0.851333             0.789861          0.061473
2960        away  0.848551             0.804787          0.043764
2959       emily  0.831529             0.903793         -0.072264
2958      heaven  0.827497             0.873439         -0.045942
2957        daba  0.824168             0.760991          0.063176
2956         did  0.819758             0.824485         -0.004727
2955         baa  0.816804             0.681281          0.135524
2954         you  0.811469                  NaN               NaN
2953       sweet  0.811352             0.786074          0.025278
2952   misbehave  0.810419             0.761707          0.048712
2951  magnifique  0.799592             0.827833         -0.028242
2950      coffee  0.795972             0.814812         -0.018840

In [99]:
coefs.sort_values('coefs_with_stopword', ascending=False).head(50)

words     coefs  coefs_with_stopword  coef_differences
2987         say  1.172162             1.550532         -0.378370
2986       night  1.150758             1.521041         -0.370283
2968        easy  0.921215             1.409904         -0.488688
2999        baby  1.612322             1.400892          0.211430
2993       white  1.265955             1.326396         -0.060442
2996        love  1.316975             1.303905          0.013070
2994         old  1.307268             1.271998          0.035270
2907        blue  0.656030             1.211287         -0.555257
2977      afraid  0.999021             1.195563         -0.196543
2981        dear  1.120327             1.190570         -0.070243
2971         met  0.935345             1.129629         -0.194284
2966        tell  0.900970             1.087209         -0.186239
2976        moon  0.998838             1.068738         -0.069900
2997         man  1.379970             1.048606          0.331364
2974         joe  0.965470             1.018108         -0.052638
2964        high  0.884754             0.984252         -0.099498
2885        near  0.600730             0.980450         -0.379719
2975       money  0.994797             0.962357          0.032440
2933        said  0.736898             0.961939         -0.225041
2970       woman  0.933002             0.951492         -0.018490
2990         day  1.223301             0.950226          0.273076
2967       happy  0.908975             0.932676         -0.023701
2927         gee  0.705370             0.931619         -0.226249
2794        long  0.449430             0.925279         -0.475850
2962        band  0.854098             0.920037         -0.065939
2943        rain  0.775431             0.911069         -0.135638
2959       emily  0.831529             0.903793         -0.072264
2916      melody  0.676834             0.879805         -0.202971
2714     chances  0.377313             0.876269         -0.498956
2958      heaven  0.827497             0.873439         -0.045942
2940       thoes  0.769394             0.872142         -0.102748
2906       tiger  0.650885             0.871594         -0.220709
2759        book  0.418645             0.864686         -0.446040
2929       fight  0.713873             0.832987         -0.119114
2978       dance  0.999631             0.831959          0.167672
2951  magnifique  0.799592             0.827833         -0.028242
2956         did  0.819758             0.824485         -0.004727
2935       loved  0.744749             0.824482         -0.079733
2938       tired  0.760042             0.822730         -0.062688
2934       frank  0.744051             0.818595         -0.074544
2950      coffee  0.795972             0.814812         -0.018840
2889        mind  0.606627             0.813478         -0.206851
2963    somebody  0.863425             0.812305          0.051121
2942        floy  0.771480             0.805916         -0.034437
2960        away  0.848551             0.804787          0.043764
2896        used  0.637138             0.801523         -0.164386
2924   hollywood  0.695493             0.795376         -0.099883
2991       shall  1.229362             0.794054          0.435308
2655        care  0.350549             0.794015         -0.443466
2961       smile  0.851333             0.789861          0.061473

In [100]:
coefs.sort_values('coefs_with_stopword').head(50)

words     coefs  coefs_with_stopword  coef_differences
0          song -1.657070            -1.962134          0.305065
57         stay -0.783806            -1.275087          0.491281
39         path -0.838538            -1.151061          0.312523
14          way -1.026359            -1.140443          0.114084
16       inside -1.011767            -1.137415          0.125648
19        belle -0.960768            -1.128200          0.167432
35      goodbye -0.863936            -1.117720          0.253784
23         wish -0.950706            -1.084684          0.133977
33        world -0.895658            -1.078975          0.183317
7          life -1.069416            -1.049800         -0.019616
30           la -0.900913            -1.024638          0.123725
9          king -1.046943            -1.012902         -0.034041
18       gaston -0.975910            -1.007493          0.031583
46    believing -0.817739            -0.987285          0.169545
106      future -0.639419            -0.985922          0.346503
28      finally -0.924334            -0.983777          0.059443
60       beauty -0.775604            -0.954688          0.179084
79       mother -0.719624            -0.953933          0.234309
66     dreaming -0.757705            -0.939747          0.182043
34          max -0.877618            -0.932609          0.054991
17        proud -1.005710            -0.924672         -0.081037
32      aladdin -0.895719            -0.921055          0.025336
26     tuppence -0.936311            -0.915926         -0.020385
24       second -0.944046            -0.915014         -0.029032
492  plantation -0.255223            -0.900816          0.645593
62    afternoon -0.767073            -0.895746          0.128673
196         air -0.483057            -0.891650          0.408592
36         chim -0.853659            -0.887293          0.033633
138        boss -0.583731            -0.884237          0.300506
25           ha -0.940213            -0.855990         -0.084223
61        ariel -0.774336            -0.847287          0.072951
51         hero -0.801683            -0.844551          0.042867
203    tomorrow -0.477343            -0.842680          0.365337
147       seein -0.560429            -0.842099          0.281669
29          ali -0.906400            -0.838822         -0.067578
40        makes -0.831353            -0.836872          0.005519
82      dancing -0.706803            -0.836242          0.129440
56        gonna -0.788030            -0.835736          0.047707
85      giselle -0.699420            -0.830262          0.130843
59         feel -0.782760            -0.822893          0.040134
22         come -0.951190            -0.812896         -0.138294
44       wonder -0.825329            -0.811415         -0.013914
96          dig -0.665314            -0.807772          0.142458
48          bop -0.808798            -0.805022         -0.003776
93     birthday -0.675337            -0.802090          0.126753
58       getcha -0.782819            -0.799774          0.016955
129  cinderelly -0.600820            -0.799207          0.198387
20         need -0.960626            -0.798915         -0.161711
131       tramp -0.597745            -0.798259          0.200514
92   understand -0.677388            -0.797558          0.120170

In [107]:
coefs.sort_values('coef_differences', ascending=False).head(50)

words     coefs  coefs_with_stopword  coef_differences
492   plantation -0.255223            -0.900816          0.645593
57          stay -0.783806            -1.275087          0.491281
1782     believe  0.062822            -0.407717          0.470539
700      humming -0.174700            -0.627555          0.452855
1411         sun -0.014634            -0.453337          0.438704
2991       shall  1.229362             0.794054          0.435308
354   unmeltable -0.331853            -0.758006          0.426153
196          air -0.483057            -0.891650          0.408592
1119       feels -0.079897            -0.451274          0.371377
203     tomorrow -0.477343            -0.842680          0.365337
504        topsy -0.247237            -0.596784          0.349547
503        turvy -0.247237            -0.596784          0.349547
106       future -0.639419            -0.985922          0.346503
599       scream -0.211913            -0.552992          0.341079
2997         man  1.379970             1.048606          0.331364
750      maurice -0.163862            -0.489822          0.325961
501         word -0.248649            -0.570600          0.321951
1735       stick  0.052269            -0.265001          0.317270
39          path -0.838538            -1.151061          0.312523
1121   christmas -0.079562            -0.391704          0.312142
2675        dark  0.359420             0.051581          0.307839
0           song -1.657070            -1.962134          0.305065
2915      cotton  0.676747             0.375332          0.301415
138         boss -0.583731            -0.884237          0.300506
2437       words  0.241921            -0.054935          0.296855
550        older -0.227543            -0.522541          0.294998
299   befuddling -0.386014            -0.679811          0.293798
1136       share -0.077636            -0.367321          0.289685
460         seen -0.274121            -0.560457          0.286336
766       making -0.161528            -0.447383          0.285854
2527       looks  0.280431            -0.003244          0.283675
147        seein -0.560429            -0.842099          0.281669
2007        sing  0.110398            -0.166625          0.277022
2990         day  1.223301             0.950226          0.273076
296         home -0.391164            -0.662729          0.271564
2778        gone  0.431183             0.159991          0.271192
835     strength -0.143892            -0.415057          0.271165
2682        dawn  0.362270             0.092155          0.270115
253     princess -0.433314            -0.698035          0.264721
2459        hand  0.249198            -0.012842          0.262039
266          hup -0.415726            -0.677132          0.261406
281       secret -0.404244            -0.662508          0.258264
774      fulfill -0.158345            -0.414026          0.255681
35       goodbye -0.863936            -1.117720          0.253784
2892      people  0.620580             0.369327          0.251253
1265     instead -0.050195            -0.294308          0.244113
2843        lady  0.518095             0.278948          0.239146
2894       cares  0.632474             0.394607          0.237867
524        color -0.239239            -0.476933          0.237694
1742      flying  0.053750            -0.183679          0.237429

In [106]:
coefs.sort_values('coef_differences').head(50)

words     coefs  coefs_with_stopword  coef_differences
2907       blue  0.656030             1.211287         -0.555257
1587  surprised  0.019847             0.564383         -0.544536
2714    chances  0.377313             0.876269         -0.498956
2968       easy  0.921215             1.409904         -0.488688
2794       long  0.449430             0.925279         -0.475850
2593     bessie  0.314168             0.773870         -0.459702
2759       book  0.418645             0.864686         -0.446040
2655       care  0.350549             0.794015         -0.443466
274       moana -0.408946             0.000000         -0.408946
31     remember -0.900538            -0.504907         -0.395631
2885       near  0.600730             0.980450         -0.379719
2987        say  1.172162             1.550532         -0.378370
2986      night  1.150758             1.521041         -0.370283
2602        hat  0.319249             0.683110         -0.363862
571     british -0.220896             0.137419         -0.358315
2678     supper  0.360543             0.717800         -0.357256
1368      leave -0.025673             0.314185         -0.339857
2153    entered  0.146852             0.475014         -0.328162
2745       tune  0.404230             0.729380         -0.325150
2477         mm  0.257507             0.581606         -0.324099
2485        won  0.261132             0.584290         -0.323158
2275     broken  0.185350             0.500503         -0.315154
53         like -0.800873            -0.487002         -0.313871
1802     behave  0.065275             0.378037         -0.312762
166        look -0.523512            -0.211122         -0.312389
1052       real -0.094379             0.212694         -0.307073
1930      stray  0.094460             0.400481         -0.306021
2595      devil  0.315449             0.620371         -0.304922
2158    country  0.148379             0.452257         -0.303878
221      change -0.462765            -0.159242         -0.303523
2752      neath  0.410496             0.710738         -0.300242
10      friends -1.042089            -0.745971         -0.296117
1443         ga -0.005626             0.285809         -0.291435
1519    morning  0.006331             0.290951         -0.284620
2547     belong  0.290425             0.571075         -0.280651
1868     forget  0.078058             0.355523         -0.277465
1694       took  0.043542             0.316457         -0.272914
2226      adore  0.172414             0.443388         -0.270974
2813    laughed  0.473634             0.742856         -0.269222
1413       true -0.014425             0.251781         -0.266206
2412         nd  0.229636             0.494758         -0.265122
633        want -0.198271             0.065145         -0.263416
1110       lies -0.081307             0.182010         -0.263317
2269   december  0.184465             0.445360         -0.260895
355       cause -0.331550            -0.071117         -0.260433
1995        fat  0.108549             0.368228         -0.259680
2567      spell  0.300531             0.558992         -0.258461
2815     rhythm  0.475996             0.734053         -0.258057
1277    promise -0.046979             0.208064         -0.255043
1744       lack  0.053890             0.306656         -0.252766